polkadot/node/network/bridge/src/rx/mod.rs

```diff
@@ -775,15 +776,29 @@ where
 					flesh_out_topology_peers(&mut authority_discovery_service, canonical_shuffling)
 						.await;
 
-				dispatch_validation_event_to_all_unbounded(
-					NetworkBridgeEvent::NewGossipTopology(NewGossipTopology {
-						session,
-						topology: SessionGridTopology::new(shuffled_indices, topology_peers),
-						local_index,
-					}),
-					ctx.sender(),
-					approval_voting_parallel_enabled,
-				);
+				if session >= newest_session {
```

Comment:
we could add a unit test for this new logic

Reply:
Done.